iCount/demultiplex.py

```diff
@@ -141,34 +141,40 @@ def _extract(reads, barcodes, mismatches=1, minimum_length=15):
         FastqEntry, experiment ID, randomer
 
     """
-    all_barcodes = len(barcodes)
-    barcode_len = len(barcodes[0])
     p2n2i = _make_p2n2i(barcodes)
-    max_votes = len(p2n2i)
-    randomer_pos = [i for i in range(barcode_len) if i not in p2n2i]
+    all_barcodes = len(barcodes)
+    # Precompute barcode length for each barcode
+    barcode_len = {i: len(brc) for i, brc in enumerate(barcodes)}
+    # Precompute max possible votes for each barcode. Equals to the number of valid nucleotides
+    max_votes = {i: len(brc.replace('N', '')) for i, brc in enumerate(barcodes)}
+    # Precompute positions of randomer nucleotides for each barcode
+    randomer_pos = {
+        i: [i for i, nuc in enumerate(brc) if nuc == 'N'] for i, brc in enumerate(barcodes)}
```

Comment:
Use other variable name for inside loop, for example `j`. Now, `i` is used in both loops. It is confusing, although correct syntactically and works as intended, at least in [python3](https://stackoverflow.com/questions/4198906/python-list-comprehension-rebind-names-even-after-scope-of-comprehension-is-thi).

Reply:
Good point, fixed.